src/JobStateChange/Repository.php

```diff
@@ -21,6 +21,23 @@ class Repository implements RepositoryInterface
     /** @var Connection */
     protected $doctrineConnection;
 
+    public function insertUsingConnection(JobStateChangeInterface $jobStateChange, Connection $connection) : RepositoryInterface
```

Comment:
I'm also open to different ways of implementing this

One alternative is to get rid of the new named connection I created, and always use the `'job'` connection. When `Data\Job` writes to this table, it's supposed to use `'job'`, and when the JSCLP reads and deletes from the table, it doesn't matter which connection it uses because it's a self-contained process.

Thoughts?

Reply:
I don't see any drawbacks to this current implementation.